### DISCUSSION TITLE
MKS-DLC32 V2.1 - ESP32-WROOM-32U

### DISCUSSION THREAD

**Discussion by drom89** [grblHAL/Controllers#3] [Q&A] [answered]
Hello, what do you think about MKS-DLC32 V2.1 - ESP32-WROOM-32U?
In terms of performance, if you compare it to BTT SKR 1.4T ?

https://github.com/makerbase-mks/MKS-DLC32

**Comment by terjeio** [comment, marked as answer]
Hard to tell without testing - and performance for what? Laser engraving?
If comms via USB then the LPC176x has faster transfer rates as the ESP32 do not have a USB peripheral.

The ESP32 driver runs under FreeRTOS - a plus or a minus? And I have to say I am not a big fan of the ESP framework and the annoying meditating guru that wakes up at seemingly random times. The FPU performance is not impressive either, but it is a bit unfair to compare that with the LPC176x as it does not have any...

Nothing beats a iMXRT1062 (Teensy 4.x) when it comes to raw performance.
